File: apprefurbished/src/main/java/com/androidcollider/easyfin/accounts/list/AccountsMVP.java
package com.androidcollider.easyfin.accounts.list;

import android.support.annotation.Nullable;

import com.androidcollider.easyfin.common.models.Account;

import java.util.List;

import io.reactivex.Flowable;

/**
 * @author Ihor Bilous
 */

public interface AccountsMVP {

    interface Model {

        Flowable<List<AccountViewModel>> getAccountList();

        Flowable<Account> getAccountById(int id);

        Flowable<Boolean> deleteAccountById(int id);
    }

    interface View {

        void setAccountList(List<AccountViewModel> accountList);

        void goToEditAccount(Account account);

        void deleteAccount();
    }

    interface Presenter {

        void setView(@Nullable AccountsMVP.View view);

        void loadData();

        void getAccountById(int id);

        void deleteAccountById(int id);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/accounts/list/AccountsMVP.kt
package com.androidcollider.easyfin.accounts.list

import com.androidcollider.easyfin.common.models.Account
import io.reactivex.rxjava3.core.Single

/**
 * @author Ihor Bilous
 */
interface AccountsMVP {
    interface Model {
        val accountList: Single<List<AccountViewModel>>
        fun getAccountById(id: Int): Single<Account>
        fun deleteAccountById(id: Int): Single<Boolean>
    }

    interface View {
        fun setAccountList(accountList: List<AccountViewModel>)
        fun goToEditAccount(account: Account)
        fun deleteAccount()
    }

    interface Presenter {
        fun setView(view: View?)
        fun loadData()
        fun getAccountById(id: Int)
        fun deleteAccountById(id: Int)
    }
}